Ports: libicu port doesn't build when added as a dependency

If you manually go to Ports/libicu and run `./package.sh` everything seems to work fine.

However, when building libicu as part of a package that has it as a dependency, it gets mixed up and tries to use the cross-compiler for the host build.

The three current packages that depend on libicu are freeciv, openttd and harfbuzz. harfbuzz is by far the easier to test this with.

So, steps to reproduce:

1) setup serenity install for i686 (for example ./Meta/serenity.sh run then exit the Qemu window) 
2) make sure that existing ports are cleaned up
   - rm -r ./Build/i686/Root/usr/Ports
   - rm -r ./Build/i686/Root/usr/local
   - cd ./Ports && git clean -xdn
   - git clean -xdf # !!! This will clean things based on the output of the xdn command above !!!
3) Build harfbuzz port
   - cd ./Ports/harfbuzz
   - ./package.sh
   
Observe that libicu fails doing the host build as such:

```
Configuring libicu!
+ mkdir -p ../host-build
+ sh -c cd ../host-build && ../source/configure && make -j12
checking for ICU version numbers... release 69.1, library 69.1, unicode version 13.0
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking whether to build debug libraries... no
checking whether to build release libraries... yes
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... configure: error: in `/home/andrew/serenity/Ports/libicu/icu/host-build':
configure: error: cannot run C compiled programs.
If you meant to cross compile, use `--host'.
See `config.log' for more details
```

Note that libicu's host-build recognizes that we are doing a host build.... BUT, the output variables still point to i686-pc-serenity-xxxx

Ports/libicu/icu/host-build/config.log
```
## ----------------- ##
## Output variables. ##
## ----------------- ##

AR='i686-pc-serenity-ar'
ARFLAGS=''
CC='i686-pc-serenity-gcc'
CFLAGS='-DU_HAVE_NL_LANGINFO_CODESET=0'
COMPILE_LINK_ENVVAR=''
CPP=''
CPPFLAGS=''
CXX='i686-pc-serenity-g++'
CXXFLAGS='-DU_HAVE_NL_LANGINFO_CODESET=0'
...
PKG_CONFIG_LIBDIR='/home/andrew/serenity/Build/i686/Root/usr/lib/pkgconfig/:/home/andrew/serenity/Build/i686/Root/usr/local/lib/pkgconfig'
PKG_CONFIG_PATH=''
PLUGINS_TRUE=''
PYTHON=''
RANLIB='i686-pc-serenity-ranlib'
```

And it does actually perform the C compiler checks with the serenity compiler
```
configure:3149: i686-pc-serenity-gcc -o conftest -DU_HAVE_NL_LANGINFO_CODESET=0   conftest.c  >&5
configure:3153: $? = 0
configure:3175: result: 
configure:3197: checking whether we are cross compiling
configure:3205: i686-pc-serenity-gcc -o conftest -DU_HAVE_NL_LANGINFO_CODESET=0   conftest.c  >&5
configure:3209: $? = 0
configure:3216: ./conftest
../source/configure: line 3218: ./conftest: No such file or directory
configure:3220: $? = 127
configure:3227: error: in `/home/andrew/serenity/Ports/libicu/icu/host-build':
configure:3229: error: cannot run C compiled programs.
If you meant to cross compile, use `--host'.
See `config.log' for more details
```

My theory is that there's something wrong with the logic for going from `target_env` to `host_env` and back, b/c the libicu package.sh does this:

```sh
configure() {
    host_env
    run mkdir -p ../host-build
    run sh -c "cd ../host-build && ../source/configure && make $makeopts"
    target_env
    run ./configure --host="${SERENITY_ARCH}-pc-serenity" $configopts
}
```

cc @gunnarbeutner :)

--- a/Ports/.port_include.sh
+++ b/Ports/.port_include.sh
@@ -4,14 +4,16 @@ set -eu
 SCRIPT="$(dirname "${0}")"
 export SERENITY_ARCH="${SERENITY_ARCH:-i686}"
 
-HOST_CC="${CC:=cc}"
-HOST_CXX="${CXX:=c++}"
-HOST_AR="${AR:=ar}"
-HOST_RANLIB="${RANLIB:=ranlib}"
-HOST_PATH="${PATH:=}"
-HOST_PKG_CONFIG_DIR="${PKG_CONFIG_DIR:=}"
-HOST_PKG_CONFIG_SYSROOT_DIR="${PKG_CONFIG_SYSROOT_DIR:=}"
-HOST_PKG_CONFIG_LIBDIR="${PKG_CONFIG_LIBDIR:=}"
+if [ -z "${HOST_CC:=}" ]; then
+    export HOST_CC="${CC:=cc}"
+    export HOST_CXX="${CXX:=c++}"
+    export HOST_AR="${AR:=ar}"
+    export HOST_RANLIB="${RANLIB:=ranlib}"
+    export HOST_PATH="${PATH:=}"
+    export HOST_PKG_CONFIG_DIR="${PKG_CONFIG_DIR:=}"
+    export HOST_PKG_CONFIG_SYSROOT_DIR="${PKG_CONFIG_SYSROOT_DIR:=}"
+    export HOST_PKG_CONFIG_LIBDIR="${PKG_CONFIG_LIBDIR:=}"
+fi
 
 DESTDIR="/"
 

--- a/Ports/curl/package.sh
+++ b/Ports/curl/package.sh
@@ -4,5 +4,11 @@ version=7.78.0
 useconfigure=true
 files="https://curl.se/download/curl-${version}.tar.bz2 curl-${version}.tar.bz2 98530b317dc95ccb324bbe4f834f07bb642fbc393b794ddf3434f246a71ea44a"
 auth_type=sha256
-depends="openssl zlib"
-configopts="--disable-ntlm-wb --with-openssl=${SERENITY_INSTALL_ROOT}/usr/local"
+depends="openssl zlib zstd"
+configopts="--disable-ntlm-wb --with-openssl=${SERENITY_INSTALL_ROOT}/usr/local --disable-symbol-hiding"
+
+install() {
+    run make DESTDIR=${SERENITY_INSTALL_ROOT} $installopts install
+    ${CC} -shared -o ${SERENITY_INSTALL_ROOT}/usr/local/lib/libcurl.so -Wl,-soname,libcurl.so -Wl,--whole-archive ${SERENITY_INSTALL_ROOT}/usr/local/lib/libcurl.a -Wl,--no-whole-archive -lzstd
+    rm -f ${SERENITY_INSTALL_ROOT}/usr/local/lib/libcurl.la
+}

--- a/Ports/gcc/package.sh
+++ b/Ports/gcc/package.sh
@@ -20,5 +20,5 @@ build() {
 
 install() {
     run make $installopts
-    run ln -s gcc "${SERENITY_INSTALL_ROOT}/usr/local/bin/cc"
+    run ln -sf gcc "${SERENITY_INSTALL_ROOT}/usr/local/bin/cc"
 }

--- a/Userland/Libraries/LibC/net/if.h
+++ b/Userland/Libraries/LibC/net/if.h
@@ -46,6 +46,14 @@ struct ifreq {
 #define ifr_hwaddr ifr_ifru.ifru_hwaddr       // MAC address
 };
 
+struct ifconf {
+    int ifc_len;
+    union {
+        void* ifc_buf;
+        struct ifreq* ifc_req;
+    };
+};
+
 unsigned int if_nametoindex(const char* ifname);
 char* if_indextoname(unsigned int ifindex, char* ifname);
 

